app/src/main/java/com/concordia/mcga/fragments/NavigationFragment.java

```diff
@@ -638,29 +637,41 @@ private void updateSearchUI() {
         if (location != null) {
             AppCompatTextView locationText = (AppCompatTextView)
                     toolbarView.findViewById(R.id.search_location_text);
-            locationText.setText(location.getName());
+            if (location instanceof Building) {
+                locationText.setText(((Building) location).getShortName());
+            } else {
+                locationText.setText(location.getName());
+            }
```

Comment:
This can be a method called getDisplayName(), which would make this slightly more compact.